Allow argument to `--threads` to be surrounded by spaces

### New feature, improvement proposal

1. Given the command:
```
mvn -V -q help:help
```
2. The command, with `.mvn/maven.config`:
```
-ntp
--threads
  1C
```
gives:
```
Apache Maven 3.9.14 (996c630dbc656c76214ce58821dcc58be960875b)
...
Java version: 21.0.10, vendor: Red Hat, Inc., runtime: ...
Default locale: en_GB, platform encoding: UTF-8
OS name: "linux" ...
```
It's fine.
3. The same command with `.mvn/maven.config`:
```
-ntp
--threads
  1
```
gives:
```
Apache Maven 3.9.14 (996c630dbc656c76214ce58821dcc58be960875b)
...
Java version: 21.0.10, vendor: Red Hat, Inc., runtime: ...
Default locale: en_GB, platform encoding: UTF-8
OS name: "linux" ...
[ERROR] Error executing Maven.
[ERROR] Invalid threads value: ' 1'. Supported are integer values.
```
4. Command could be also used w/o `maven.config` like so:
```
mvn -V -q help:help -T ' 1C'             # a
mvn -V -q help:help -T ' 1'              # b
```
which also works for case _a_ and doesn't for _b_.
5. Having leading or trailing space in option gives something like
```
org.apache.commons.cli.ParseException: Unrecognized maven.config file entries: [ -ntp ]
```
or
```
Unable to parse maven.config file options: Unrecognized option: -ntp
```
which might come from `commons-cli`. 

But as parsing threads value is done in Maven - could it be always trimmed first?

This is for `Float.parseFloat` (`valueOf`): https://docs.oracle.com/javase/8/docs/api/java/lang/Float.html#valueOf-java.lang.String-:
> Leading and trailing whitespace characters in `s` are ignored. Whitespace is removed as if by the [String.trim()](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#trim--) method; that is, both ASCII space and control characters are removed. **The rest** of `s` should constitute a FloatValue ...

But nothing like that happens in `Integer.parseInt`.

--- a/maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java
+++ b/maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java
@@ -1477,17 +1477,17 @@ public class MavenCli {
             }
         } else {
             try {
-                int threads = Integer.parseInt(originalThreadConfiguration);
+                int threads = Integer.parseInt(threadConfiguration);
 
                 if (threads <= 0) {
                     throw new IllegalArgumentException(
-                            "Invalid threads value: '" + originalThreadConfiguration + "'. Value must be positive.");
+                            "Invalid threads value: '" + threadConfiguration + "'. Value must be positive.");
                 }
 
                 return threads;
             } catch (NumberFormatException e) {
                 throw new IllegalArgumentException(
-                        "Invalid threads value: '" + originalThreadConfiguration + "'. Supported are integer values.");
+                        "Invalid threads value: '" + threadConfiguration + "'. Supported are integer values.");
             }
         }
     }

--- a/maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java
+++ b/maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java
@@ -1453,7 +1453,8 @@ public class MavenCli {
         return request;
     }
 
-    int calculateDegreeOfConcurrency(String threadConfiguration) {
+    int calculateDegreeOfConcurrency(String originalThreadConfiguration) {
+        String threadConfiguration = originalThreadConfiguration.trim();
         if (threadConfiguration.endsWith("C")) {
             threadConfiguration = threadConfiguration.substring(0, threadConfiguration.length() - 1);
 
@@ -1476,17 +1477,17 @@ public class MavenCli {
             }
         } else {
             try {
-                int threads = Integer.parseInt(threadConfiguration);
+                int threads = Integer.parseInt(originalThreadConfiguration);
 
                 if (threads <= 0) {
                     throw new IllegalArgumentException(
-                            "Invalid threads value: '" + threadConfiguration + "'. Value must be positive.");
+                            "Invalid threads value: '" + originalThreadConfiguration + "'. Value must be positive.");
                 }
 
                 return threads;
             } catch (NumberFormatException e) {
                 throw new IllegalArgumentException(
-                        "Invalid threads value: '" + threadConfiguration + "'. Supported are integer values.");
+                        "Invalid threads value: '" + originalThreadConfiguration + "'. Supported are integer values.");
             }
         }
     }

--- a/maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java
+++ b/maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java
@@ -92,6 +92,8 @@ public class MavenCliTest {
         assertEquals(1, cli.calculateDegreeOfConcurrency("0.0001C"));
         assertEquals(1 * cpus, cli.calculateDegreeOfConcurrency(" 1C"));
         assertEquals(1 * cpus, cli.calculateDegreeOfConcurrency(" 1C "));
+        assertEquals(1, cli.calculateDegreeOfConcurrency(" 1"));
+        assertEquals(1, cli.calculateDegreeOfConcurrency(" 1 "));
         assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("-2.2C"));
         assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("0C"));
     }

--- a/maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java
+++ b/maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java
@@ -90,6 +90,7 @@ public class MavenCliTest {
         int cpus = Runtime.getRuntime().availableProcessors();
         assertEquals((int) (cpus * 2.2), cli.calculateDegreeOfConcurrency("2.2C"));
         assertEquals(1, cli.calculateDegreeOfConcurrency("0.0001C"));
+        assertEquals(1 * cpus, cli.calculateDegreeOfConcurrency(" 1C"));
         assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("-2.2C"));
         assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("0C"));
     }

--- a/maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java
+++ b/maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java
@@ -82,6 +82,10 @@ public class MavenCliTest {
         assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("2C2"));
         assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("CXXX"));
         assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("XXXC"));
+        assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("1c"));
+        assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency(" 1c"));
+        assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("1c "));
+        assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency(" 1c "));
 
         int cpus = Runtime.getRuntime().availableProcessors();
         assertEquals((int) (cpus * 2.2), cli.calculateDegreeOfConcurrency("2.2C"));

--- a/maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java
+++ b/maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java
@@ -91,6 +91,7 @@ public class MavenCliTest {
         assertEquals((int) (cpus * 2.2), cli.calculateDegreeOfConcurrency("2.2C"));
         assertEquals(1, cli.calculateDegreeOfConcurrency("0.0001C"));
         assertEquals(1 * cpus, cli.calculateDegreeOfConcurrency(" 1C"));
+        assertEquals(1 * cpus, cli.calculateDegreeOfConcurrency(" 1C "));
         assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("-2.2C"));
         assertThrows(IllegalArgumentException.class, () -> cli.calculateDegreeOfConcurrency("0C"));
     }

